src/watchdog/src/watchdog.py

```diff
@@ -217,20 +217,20 @@ def collect_healthz(gauge, histogram, service_name, address, port, url):
         gauge.add_metric([service_name, error, address], 1)
 
 
-def collect_k8s_componentStaus(k8s_gauge, api_server_ip, api_server_port, nodesJsonObject):
+def collect_k8s_componentStaus(k8s_gauge, api_server_scheme, api_server_ip, api_server_port, nodesJsonObject, ca_path, headers):
     collect_healthz(k8s_gauge, api_healthz_histogram,
-            "k8s_api_server", api_server_ip, api_server_port, "/healthz")
+            "k8s_api_server", api_server_scheme, api_server_ip, api_server_port, "/healthz", ca_path, headers)
     collect_healthz(k8s_gauge, etcd_healthz_histogram,
-            "k8s_etcd", api_server_ip, api_server_port, "/healthz/etcd")
+            "k8s_etcd", api_server_scheme, api_server_ip, api_server_port, "/healthz/etcd", ca_path, headers)
 
     # check kubelet
     nodeItems = nodesJsonObject["items"]
 
     for name in nodeItems:
         ip = name["metadata"]["name"]
-
+        
         collect_healthz(k8s_gauge, kubelet_healthz_histogram,
-            "k8s_kubelet", ip, 10255, "/healthz")
+            "k8s_kubelet", "http", ip, 10255, "/healthz", None, None)
```

Comment:
where do you detect whether this should be "http" or "https"?

Reply:
For on-prem existing PAI it use http. 

For AKS I haven't find a method to access this kubelet port. I also suggest this metrics is better to be skipped for pai on AKS, as this responsibility should be taken by AKS and not PAI.

we could add a pending issue for future test.

Reply:
in this case you will need to discuss with the team for this decision before submitting a PR.

Reply:
Not aware this before, so not mentioned. Currently i haven't make any decision. and keep previous code as much as possible

Reply:
@xudifsd , kubelet url access issue created and assigned to you, needs your help to investigate, thanks. 
https://github.com/Microsoft/pai/issues/1878